tests/integration/test_data/rpm_test_repo_file/cachi2.repo

```diff
@@ -0,0 +1,3 @@
+[releases]
```

Comment:
Can we make this slightly more complex by testing the following additionally here?
- multiple repo sections generated
- deps which are unaffiliated with any official repo (i.e. one is generated for them)?

The latter will require some mocking of the random repo name string.

Reply:
Done. We can't really mock any Cachi2 method in the integration test, but I am replacing the random string with regex so that we can actually cover this case in the test.

I don't quite like the way the code is, but maybe we can improve it a bit after your `ConfigParser` change goes in.

Reply:
You're right, we can't! One note that I'm going to make although it won't change anything right now is the fact that we're assuming the file is going to be generated always the same way because right now our code sorts the output before dumping which is technically totally unnecessary and is also proposed to be dropped with #522 where as you mention `ConfigParser` is suggested to come into play which means we'll have to compare the file on the object level rather than 1:1 content level, so this is just a NB for me to update the PR in question when this is merged. 

> I don't quite like the way the code is, but maybe we can improve it a bit after your ConfigParser change goes in.

we could drop the need for regex handling and tackle this in the `ConfigParser` instance but I doubt it will make the code that much better, it would definitely make it longer, we'll see.